### ISSUE TITLE
[OpenMP][AMDGPU] OpenMP offload errors with O0 and save-temps

### ISSUE THREAD

**Issue by markdewing** [llvm/llvm-project#71594]
Use the reproducer from #71593.

Add "-v --save-temps" to the compile line.  Get an error:
```
a.out.amdgcn-amd-amdhsa.gfx900.s:100638:27: error: invalid operand (violates constant bus restrictions)
        v_writelane_b32 v7, s24, m0
```

Add "-O0" (and no "--save-temps") and get an assertion failure:
```
clang-linker-wrapper: llvm-project/llvm/lib/Target/AMDGPU/AMDGPUResourceUsageAnalysis.cpp:154: virtual bool llvm::AMDGPUResourceUsageAnalysis::runOnModule(llvm::Module&): Assertion `MF && "function must have been generated already"' failed.
```

Full outputs from the errors attached

[out_amd_O0.txt](https://github.com/llvm/llvm-project/files/13289882/out_amd_O0.txt)
[out_amd_save_temps.txt](https://github.com/llvm/llvm-project/files/13289883/out_amd_save_temps.txt)

**Comment by arsenm** [comment]
I already know why the second assert fails, I haven't seen the first assembler error before

**Comment by shiltian** [comment]
The first one was fixed by 6a21e00. Not sure if the second one has been fixed. If so, we can close the issue.

**Comment by jhuber6** [comment]
> The first one was fixed by [6a21e00](https://github.com/llvm/llvm-project/commit/6a21e00e397648141ed36aae4bd958efa09908f3). Not sure if the second one has been fixed. If so, we can close the issue.

The second one is most likely https://github.com/llvm/llvm-project/issues/64863, I don't think it's been fixed yet.

**Comment by arsenm** [comment]
I think we can close this, we don't need multiple issues open for the unreached in SCC order function issue
